server/src/main/java/org/opensearch/action/get/TransportGetAction.java

```diff
@@ -90,14 +93,21 @@ protected boolean resolveIndex(GetRequest request) {
 
     @Override
     protected ShardIterator shards(ClusterState state, InternalRequest request) {
+        String preference = request.request().preference();
+        // route realtime GET requests when segment replication is enabled to primary shards,
+        // iff there are no other preferences/routings enabled for routing to a specific shard
+        if (state.getMetadata()
+            .index(request.concreteIndex())
+            .getSettings()
+            .get(IndexMetadata.SETTING_REPLICATION_TYPE)
+            .equals(ReplicationType.SEGMENT.toString())
+            && request.request().realtime()
+            && request.request().routing() == null
+            && request.request().preference() == null) {
+            preference = Preference.PRIMARY.type();
```

Comment:
1. Move this to a try.catch to ensure all null checks are handled. 
2. What if the preference was already overridden to `PRIMARY`

Reply:
Thanks for the review @Bukhtawar! In the case of (2), we will honor the preference to `PRIMARY` that user has set.